include/ebpf_extension.h

```diff
@@ -91,11 +91,21 @@ typedef struct _ebpf_extension_program_dispatch_table
     ebpf_program_batch_end_invoke_function_t ebpf_program_batch_end_invoke_function;
 } ebpf_extension_program_dispatch_table_t;
 
-typedef struct _ebpf_extension_data
+typedef struct _ebpf_extension_data_v1
 {
     ebpf_extension_header_t header;
     const void* data;
-} ebpf_extension_data_t;
+} ebpf_extension_data_v1_t;
+
+typedef struct _ebpf_extension_data_v2
+{
+    ebpf_extension_header_t header;
+    const void* data;
+    size_t data_size;
```

Comment:
the extensions in the ebpf repo should be updated to use the correct `data_size` field and should probably add version checks.

Reply:
Done.